### PR TITLE
a rough implementation of enhancement described in #102

### PR DESCRIPTION
**Pull request by parmentelat**
followup for #102 

before exiting reveal mode, this patch computes
the notebook index of the first cell in the currently displayed slide
that cell then gets selected and focused when returning to normal
notebook mode

this first attempt implements this behaviour unconditionally,
i.e. regardless of `start_slideshow_at`
I am not convinced that it should depend on that setting anyway
point being that it is rather straightforward to go to the beginning or end 
of the presentation when in normal notebook mode, and generally not critical 

this is of course open to further discussion though

### CODE REVIEW

**Comment by parmentelat** [comment]
As far as testing:
* I have only been able to try it out in my own environment, it seems to be working as I expect it to, at least
* I have not been able to spot any other tests to run, please provide instructions if relevant

**Comment by parmentelat** [comment]
One last thing, as compared to what the original issue was requesting:
* I do not even look at what the 'selected' cell is in the notebook when reveal is turned off
* this is because most of the time there is no such selection
* instead I try to locate  the (first cell in the) slide that is currently displayed

**Comment by damianavila** [comment]
Thanks for the implementation, I will try to take a look as soon as I can.

File: rise/static/main.js [hunk 1]
```diff
@@ -483,13 +531,17 @@ function revealMode() {
     buttonHelp();
     $('#maintoolbar').addClass('reveal_tagging');
   } else {
+    var current_cell_index = reveal_cell_index(IPython.notebook);
```
Comment (damianavila) [inline rise/static/main.js]: You custom function is not necessary, I think... you can use something like this:
```javascript
var current_cell_index = IPython.notebook.get_selected_index()
```

File: rise/static/main.js [hunk 2]
```diff
@@ -483,13 +531,17 @@ function revealMode() {
...
     Remover(config);
     setupKeys("notebook_mode");
     $('#exit_b').remove();
     $('#help_b').remove();
     $('#maintoolbar').removeClass('reveal_tagging');
     // Workaround... should be a better solution. Need to investigate codemirror
     fixCellHeight();
+    // select and focus on current cell
+    IPython.notebook.select(current_cell_index);
+    IPython.notebook.get_selected_cell().ensure_focused();
```
Comment (damianavila) [inline rise/static/main.js]: I don't thing this line is necessary... selecting the cell by index in the previous should be enough, I think.

**Comment by damianavila** [comment]
Sorry for the delay in the review @parmentelat 
I left some comments. 
If you can not work on this further, just let me know. Thanks!

**Comment by parmentelat** [comment]
Hi

It's been a long time so things are not 100% clear in my memory, but...

I did try to use jupyter's notion of the selected cell at first; 
the code was much simpler indeed, but, in my experience, it's very easy to move inside the slideshow and to *not* have the selected cell - in the jupyter sense - be present in the slice that you are watching;

so I found that relyong on jupyter's selected cell was definitey *not* what I was after most of the time; OTOH since I wrote it this code gives me complete satisfaction

I have some limited availability to work on this, but before you make any decision please try both approaches in real situation

either way, let me know what I can do

thanks for your feedback

**Comment by damianavila** [comment]
>it's very easy to move inside the slideshow and to not have the selected cell - in the jupyter sense - be present in the slice that you are watching;

Interesting...

>so I found that relying on jupyter's selected cell was definitey not what I was after most of the time; OTOH 

OK, now with the context I better get the idea of your PR and I like it!
Sorry to not read your message above... you were clear on it, but I just went to read the implementation :wink:

OK, I will correct any issue in your approach (if I find something) and merge this one.

Thank you!!

File: rise/static/main.js (hunk 2 above)
Comment (damianavila) [inline rise/static/main.js]: OK, now testing the thing a little bit more I see why you used this line.

**Comment by damianavila** [comment]
LGTM, merging! Thanks @parmentelat !!

**Pull request merged**